drivers/soundwire/bus.c

```diff
@@ -24,9 +24,14 @@ int sdw_bus_master_add(struct sdw_bus *bus, struct device *parent,
 	struct sdw_master_prop *prop = NULL;
 	int ret;
 
-	if (!bus->dev) {
-		pr_err("SoundWire bus has no device\n");
-		return -ENODEV;
+	if (!bus)
```

Comment:
I don't see any changes in the way `sdw_bus_master_add()` gets called after this patch. Is this something that changes with this patch, that now this function can be called with `bus == NULL`? Or was it a but that it didn't check for it? If it was a bug, I think a separate patch is needed.

Reply:
> I don't see any changes in the way `sdw_bus_master_add()` gets called after this patch. Is this something that changes with this patch, that now this function can be called with `bus == NULL`? Or was it a but that it didn't check for it? If it was a bug, I think a separate patch is needed.

Thanks, I will update it.